src/main/org/codehaus/groovy/transform/stc/StaticTypeCheckingSupport.java

```diff
@@ -1574,9 +1560,9 @@ private static boolean equalIncludingGenerics(ClassNode orig, ClassNode copy) {
         if (orig.isGenericsPlaceHolder()!=copy.isGenericsPlaceHolder()) return false;
         if (!orig.equals(copy)) return false;
         GenericsType[] gt1 = orig.getGenericsTypes();
-        GenericsType[] gt2 = orig.getGenericsTypes();
-        if ((gt1==null || gt2==null) && gt1!=gt2) return false;
+        GenericsType[] gt2 = copy.getGenericsTypes();
         if (gt1==gt2) return true;
+        if (gt1 == null || gt2 == null) return false;
```

Comment:
I personally don`t like this reordering of checks here. For me the intend more clear as it was before (of course minus  the orig/copy bug)